codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java

```diff
@@ -1159,8 +1160,14 @@ private static boolean loadDataMultipartOptimized(ByteBuf undecodedChunk, String
         final byte[] bdelimiter = delimiter.getBytes(httpData.getCharset());
         int posDelimiter = HttpPostBodyUtil.findDelimiter(undecodedChunk, startReaderIndex, bdelimiter, true);
         if (posDelimiter < 0) {
-            // Not found but however perhaps because incomplete so search LF or CRLF
-            posDelimiter = HttpPostBodyUtil.findLineBreak(undecodedChunk, startReaderIndex);
+            // Not found but however perhaps because incomplete so search LF or CRLF from the end within delimiter
+            // possible partial last bytes
+            int lastPosition = undecodedChunk.readableBytes() - bdelimiter.length;
+            if (lastPosition < 0) {
+                // Not enough bytes but can still try to find CRLF
+                lastPosition = 0;
+            }
+            posDelimiter = HttpPostBodyUtil.findLastLineBreak(undecodedChunk, startReaderIndex  + lastPosition);
```

Comment:
So… we didn't find a complete delimiter, _but_ if we see a CRLF within a delimiters length of the end, then we _might_ have a delimiter that straddles two read calls i.e. two buffers. And that means we can't be sure that we won't have a delimiter in that byte range, so we can only add the content prior to the CRLF to the HttpData. And then in the next call, the data we didn't read here will be carried over, and get more added to it, and then we'll be able to tell if there's really a delimiter there or not.

Is my understanding correct?

Reply:
Correction done